### ISSUE TITLE
Should use long long instead of long or int to represent steps

### ISSUE THREAD

**Issue by HanatoK** [Colvars/colvars#239]
My colleague tried to restart from a ABF simulation which the `step` in the state file was larger than the maximum value of `long` data type, and she encountered a bunch of lines containing "Prepared sample and gradient buffers at step XXX" in the NAMD standard output.
I took a look at the code in colvarbias_abf.cpp:
```
  // Prepare for the first sharing.
  if (shared_last_step < 0) {
    // Copy the current gradient and count values into last.
    last_gradients->copy_grid(*gradients);
    last_samples->copy_grid(*samples);
    shared_last_step = cvm::step_absolute();
    cvm::log("Prepared sample and gradient buffers at step "+cvm::to_str(cvm::step_absolute())+".");
  }
```
The `shared_last_step` is a variable of `int` type, and when the step is larger than `2,147,483,647` then it's always less than zero. `2,147,483,647` steps with a 2.0 fs timestep approximates 4.3 microseconds.
Sometimes we need to run simulations more than 4.3 microseconds, so I think use `long long` instead of `long` or `int` as the type of `shared_last_step`, `cvm::step_absolute()` and step-related variables is better.

**Comment by jhenin** [comment]
We could switch at the Colvars level, but keep in mind that NAMD itself stores the time step number as an `int`. See eg. this thread: https://www.ks.uiuc.edu/Research/namd/mailing_list/namd-l.2016-2017/0975.html
To fix that you need to reset periodically to 0 with `firstTimeStep`.

That said, Colvars keeps its own record of the time step through restart files, so moving to a longer type could work independently from the back-end. Thoughts, @giacomofiorin ?

**Comment by giacomofiorin** [comment]
Yeah.  I'd have changed to long long but was waiting to see what specific
type would NAMD converge to (e.g. int64).  Some compilers may nag if they
are spelled differently.  However, even after the introduction of 64bit for
global atom indices that didn't happen yet.  Best to handle it in Colvars
at this point.

We are currently honoring firstTimeStep at setup if it is larger than zero,
so definitely we need to address the case of overflow in NAMD itself.  For
example, by warning the user if the Colvars state file has a larger number
than NAMD can represent for firstTimeStep.  Not sure if this should be an
error.

Also, the type of all step numbers in the subclasses must change as well,
so best to use a typedef here.

**Comment by HanatoK** [comment]
Fix by #241
